### PR TITLE
Add a scraper for Randolph County, NC

### PR DESCRIPTION
**Pull request by ralmarzoog**

### CODE REVIEW

File: recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    def _create_person(self, ingest_info, personElem):
+        first_name = self._get_single_element(personElem, "nf")
+        middle_name = self._get_single_element(personElem, "nm")
+        race_and_ethnicity = self._get_single_element(personElem, "race")
+        street_address = self._get_single_element(personElem, "street")
+        city_state_zip = self._get_single_element(personElem, "csz")
+
+        person = ingest_info.create_person(
+            surname=self._get_single_element(personElem, "nl"),
+            given_names=' '.join([first_name, middle_name]),
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py]: nit: the ingest info has a `middle_names` field too, shouldn't make a big difference!

**Comment by ralmarzoog** [comment]
I did try to use the data extractor and at first it didn't work because the data was returned as XML w/o the labels. Tried again and got the data as html and it somewhat works but it gets confused with the way the arrest subtables are set up if the person has multiple arrests; e.g. it'll start mapping charging_entity to the next person in the list's name and will put the charge date and name in separate charge entities.

File: recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py (hunk 1 above)
Comment (ralmarzoog) [reply recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py]: Oops don't know how I missed that. Done.

File: recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    @staticmethod
+    def _get_single_element(root, path):
+        """ Returns the element at the path in relation to the given root.
+            The path should point to a single element. If it does not,
+            None will be returned instead.
+        """
+        elem = root.xpath(".//" + path)
+        # pylint: disable=literal-comparison
+        if len(elem) is not 1:
```
[the remark this answers is not in the document]
Comment (ralmarzoog) [reply recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py]: Yup, thanks!

File: recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            person = self._create_person(ingest_info, personElem)
+            self._create_booking(person, personElem)
+
+        return ScrapedData(ingest_info=ingest_info, persist=True)
+
+    def get_initial_task(self) -> Task:
+        return Task(endpoint=self.get_region().base_url,
+                    task_type=TaskType.SCRAPE_DATA)
+
+    def get_more_tasks(self, content, task: Task) -> List[Task]:
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py]: you can delete this as it's no longer abstract!

File: recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            for charge_elem in self._get_repeated_element(arrest_elem, "of"):
+                self._create_charge(charge_elem, arrest_elem, booking)
+
+        if not arrest_elem_list:
+            return
+
+        # Because we can't have multiple arrests for a single booking, just
+        # record the first arresting officer's info to avoid losing all
+        # officer data
+        # Because we can't have multiple arrests for a single booking, just
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_nc_randolph/us_nc_randolph_scraper.py]: this comment is duplicated

**Comment by jamwalla** [comment]
merged this, thank you!!

**Pull request merged**